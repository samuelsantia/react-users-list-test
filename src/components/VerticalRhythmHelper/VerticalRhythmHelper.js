import React from 'react';
import styled from 'styled-components/macro';

/** shows vertical base grid */
export const VerticalRhythmHelperBase = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  pointer-events: none;
  user-select: none;
  background: linear-gradient(rgba(0, 119, 179, 0.2) 1px, transparent 1px) left
    top / var(--scale-rhythm) var(--scale-rhythm);
`;

/**
 * shows vertical base grid
 * dont renders in production environments
 */
export default function VerticalRhythmHelper(props) {
  if (process.env.NODE_ENV === 'production') {
    return null;
  }
  return <VerticalRhythmHelperBase {...props} />;
}
