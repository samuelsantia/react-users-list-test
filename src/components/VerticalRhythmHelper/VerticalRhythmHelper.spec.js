import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import VerticalRhythmHelper from './VerticalRhythmHelper';

let origEnv = process.env.NODE_ENV;

afterEach(() => {
  process.env.NODE_ENV = origEnv;
});

describe('<VerticalRhythmHelper />', () => {
  it('should render', () => {
    const { container } = render(<VerticalRhythmHelper />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render if NODE_ENV is `production`', () => {
    process.env.NODE_ENV = 'production';
    const { container } = render(<VerticalRhythmHelper />);

    expect(container.firstChild).toBeNull();
  });
});
