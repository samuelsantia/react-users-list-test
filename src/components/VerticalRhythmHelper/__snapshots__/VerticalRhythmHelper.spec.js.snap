// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalRhythmHelper /> should render 1`] = `
.c0 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: linear-gradient(rgba(0,119,179,0.2) 1px,transparent 1px) left top / var(--scale-rhythm) var(--scale-rhythm);
}

<div
  class="c0"
/>
`;
